sdks/python/apache_beam/typehints/native_type_compatibility.py

```diff
@@ -107,7 +107,7 @@ def _match_is_exactly_iterable(user_type):
 def match_is_named_tuple(user_type):
   return (
       _safe_issubclass(user_type, typing.Tuple) and
-      hasattr(user_type, '_field_types'))
+      hasattr(user_type, '__annotations__'))
```

Comment:
Are we sure this won't lead to false positives? That is, are we sure the only way a tuple subclass will have an `__annotations__` attribute is when it's a `NamedTuple`?

Reply:
Sounds good, thank you!